server/src/main/scala/io/delta/common/SnapshotChecker.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.common
+
+import io.delta.sharing.server.DeltaSharingUnsupportedOperationException
+import io.delta.sharing.server.actions.{ColumnMappingTableFeature, DeltaAction}
+
+
+object SnapshotChecker {
+
+  /**
+   * Assert that the client is up to date with the protocol and allowed
+   * to read the table that is using this Snapshot's `protocol`.
+   */
+  def assertProtocolRead(tblMinReaderVersion: Int): Unit = {
```

Comment:
Actually we need to add test for these functions.

Reply:
How can I write a test where the client is not up to date with the protocol? Would I need to write a client test for this in Python?